### PR TITLE
Workaround for tuple axis arg for masked arrays

### PR DESCRIPTION
**Pull request by vidartf**
While numpy's methods `mean`, `sum` etc. accept tuples for their `axis` argument, `numpy.ma.MaskedArray` does not (yet?). So for that reason, I propose this workaround for those data types.

Note, the function signature includes the `out` keyword in order to be complete w.r.t that PR. This workaround should therefore be considered blocked by that PR.

### CODE REVIEW

**Comment by vidartf** [comment]
With #424 merged, this is now ready for review.

**Comment by to266** [comment]
At least one test?

File: hyperspy/signal.py [hunk 1]
```diff
@@ -3872,6 +3872,18 @@ def _remove_axis(self, axes):
                 name="Scalar",
                 navigate=False,)
 
+    def _ma_workaround(self, s, function, axes, ar_axes, out):
+        # TODO: Remove if and when numpy.ma accepts tuple `axis`
+        intermediate = self
+        for a in ar_axes[:-1]:
```
Comment (to266) [inline hyperspy/signal.py]: maybe `reversed`? Haven't tested it though..

**Comment by vidartf** [comment]
Good point, since it doesn't work :)

**Comment by vidartf** [comment]
PS: When running tests locally, I get the following error:
```python
======================================================================
ERROR: hyperspy.tests.signal.test_tools.TestOutArg.test_indexmax
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\prog\WinPython-64bit-2.7.10.2\python-2.7.10.amd64\lib\site-packages\nose\case.py", line 197, in runTest
    self.test(*self.arg)
  File "c:\github\hyperspy\hyperspy\tests\signal\test_tools.py", line 473, in test_indexmax
    self._run_single(self.s.indexmax, self.s, dict(axis=0))
  File "c:\github\hyperspy\hyperspy\tests\signal\test_tools.py", line 435, in _run_single
    r = f(out=s1, **kwargs)
  File "c:\github\hyperspy\hyperspy\signal.py", line 4288, in indexmax
    out=out)
  File "c:\github\hyperspy\hyperspy\signal.py", line 3908, in _apply_function_on_data_and_remove_axis
    function(self.data, axis=ar_axes, out=out.data)
TypeError: argmax() got an unexpected keyword argument 'out'
```

Seems to be that `argmax` only got the `out` arg in numpy 1.10, and I'm still at 1.9. Just a heads up!

**Comment by vidartf** [comment]
Tests have been introduced and behavior fixed, so should be ready for review. It currently only tests for `sum` and `mean`. I *think* it should work for the others as well, but didn't know how to test for those without reproducing the implementation in the test code.

File: hyperspy/tests/signal/test_tools.py [hunk 2]
```diff
@@ -528,3 +528,25 @@ def test_histogram_axis_changes(self):
         assert_array_equal(h1.data, h2.data)
         nt.assert_equal(h1.axes_manager[-1].size,
                         h2.axes_manager[-1].size,)
+
+    def test_masked_array_mean(self):
+        s = self.s
+        mask = (s.data > 0.5)
+        s.data = np.ones_like(s.data)
```
Comment (to266) [inline hyperspy/tests/signal/test_tools.py]: I would rather you used something like `np.arange(s.data.size).reshape(s.data.shape)`. When all elements in an array are 1, testing if a particular subset has a mean of 1 is not very indicative that the subset was selected correctly...

Comment (vidartf) [inline hyperspy/tests/signal/test_tools.py]: Implemented.

Comment (to266) [inline hyperspy/tests/signal/test_tools.py]: Tests fail though, at least in travis

Comment (vidartf) [inline hyperspy/tests/signal/test_tools.py]: Hah, I forgot that the mask is based on random data.

Comment (vidartf) [inline hyperspy/tests/signal/test_tools.py]: Added a fixed seed, and updated the reference data, so now it should be OK.

**Pull request merged**